examples/event_handling/EventHandling.cpp

```diff
@@ -0,0 +1,224 @@
+                                      { m_log.emplace_back("1: Touch Began: " + vec2ToString(touchBegan.position)); });
+
+                // To handle unhandled events, just add the following lambda to the set of handlers
+                // [&](const auto&) { m_log.emplace_back("1: Other Event"); }
+            }
+            else if (handlerType == 2)
+            {
+                // Generic visitation
+                // A generic callable is provided that can differentiate by deduced event type
+                m_window.handleEvents(
+                    [&](auto&& event)
+                    {
+                        // Remove reference and cv-qualifiers
+                        using T = std::decay_t<decltype(event)>;
+
+                        if constexpr (std::is_same_v<T, sf::Event::Closed>)
+                        {
+                            m_window.close();
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::KeyPressed>)
+                        {
+                            m_log.emplace_back("2: Key Pressed: " + sf::Keyboard::getDescription(event.scancode));
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::MouseMoved>)
+                        {
+                            m_log.emplace_back("2: Mouse Moved: " + vec2ToString(event.position));
+                        }
+                        if constexpr (std::is_same_v<T, sf::Event::MouseButtonPressed>)
+                        {
+                            m_log.emplace_back("2: Mouse Pressed");
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::TouchBegan>)
+                        {
+                            m_log.emplace_back("2: Touch Began: " + vec2ToString(event.position));
+                        }
+                        else
+                        {
+                            // All unhandled events will end up here
+                            // m_log.emplace_back("2: Other Event");
+                        }
+                    });
+            }
+            else if (handlerType == 3)
+            {
+                // Forward to other callable
+                // In this case we forward it to our handle member functions
+                // we defined for the concrete event types we want to handle
+                // When choosing this method a default "catch-all" handler must
+                // be available for unhandled events to be forwarded to
+                // If you don't want to provide an empty "catch-all" handler
+                // you will have to make sure (e.g. via if constexpr) that this
+                // lambda doesn't attempt to call a member function that doesn't exist
+                m_window.handleEvents([this](const auto& event) { handle(event); });
+            }
+
+            // Switch to next form of event handling
+            handlerType = (handlerType + 1) % 4;
```

Comment:
I expected some input switch to change implementation, which would also allow/invite experimentation and comparison from user side. With constant switching it's kind of random which handler type is active at any given time, making it way less predictable/reproducable.

Reply:
Fixed.